client/src/transaction/api/client.rs

```diff
@@ -0,0 +1,351 @@
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_GET_BLOCK: Method<GetBlockRequest, Block> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/GetBlock",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_GET_BLOCK_BY_HASH: Method<GetBlockByHashRequest, Block> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/GetBlockByHash",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_GET_TX: Method<GetTxRequest, TxResult> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/GetTx",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_GET_TX_BY_BLOCK_HASH: Method<GetTxByBlockHashRequest, TxResult> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/GetTxByBlockHash",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_GET_TXS: Method<GetTxsRequest, TxnBatch> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/GetTxs",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_QUERY_TX: Method<QueryTxRequest, TxResult> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/QueryTx",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_QUERY_TXS: Method<QueryTxsRequest, Vec<TxResult>> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/QueryTxs",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_WAIT_BLOCK_INDEXED: Method<WaitBlockIndexedRequest, ()> = Method {
+    ty: MethodType::Unary,
+    name: "/oasis-core.RuntimeClient/WaitBlockIndexed",
+    req_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+    resp_mar: Marshaller {
+        ser: grpc::cbor_encode,
+        de: grpc::cbor_decode,
+    },
+};
+
+const METHOD_WATCH_BLOCKS: Method<RuntimeId, AnnotatedBlock> = Method {
```

Comment:
Could you use the macro for defining these Methods (and others in api/)? Most attributes seem to be the same.

Reply:
Good idea, done.